### PR TITLE
Os rev git sha1

### PR DESCRIPTION
**Pull request by jforissier**

### CODE REVIEW

**Comment by etienne-lms** [comment]
`Reviewed-by: Etienne Carriere <etienne.carriere@linaro.org>`
nice.

**Comment by jenswi-linaro** [comment]
I think we're missing a way to disable this for production builds.

**Comment by jforissier** [comment]
> I think we're missing a way to disable this for production builds.

Why would you want to do that?

**Comment by jenswi-linaro** [comment]
This sha-sum is basically a debug feature, but it also uniquely identifies the source code. In production builds you may want more control over what information you leak to normal world.

**Comment by jforissier** [comment]
OK, how about: `CFG_OS_REV_REPORTS_GIT_SHA1 ?= y`
?

**Comment by jenswi-linaro** [comment]
Sounds good

**Comment by jforissier** [comment]
Update

File: core/arch/arm/tee/entry_fast.c [hunk 1]
```diff
@@ -242,4 +242,7 @@ void __weak tee_entry_get_os_revision(struct thread_smc_args *args)
 {
 	args->a0 = CFG_OPTEE_REVISION_MAJOR;
 	args->a1 = CFG_OPTEE_REVISION_MINOR;
+#ifdef CFG_OS_REV_REPORTS_GIT_SHA1
+	args->a2 = TEE_IMPL_GIT_SHA1;
+#endif
```
Comment (jenswi-linaro) [inline core/arch/arm/tee/entry_fast.c]: Wouldn't it be more friendly to assign zero in case this feature isn't enabled, instead of leaving it unchanged?

Comment (etienne-lms) [reply core/arch/arm/tee/entry_fast.c]: If so, maybe use `CFG_OS_REV_REPORTS_GIT_SHA1` when generating the SHA1 (fill it with 0's).
Nice though @jenswi-linaro about product considerations.

Comment (jforissier) [reply core/arch/arm/tee/entry_fast.c]: Well, the caller can initialize it to zero or whatever special value to mean "unspecified", and actually it has to if it wants to be compatible with the previous implementation. So I'm not sure it is really needed.

Comment (jenswi-linaro) [reply core/arch/arm/tee/entry_fast.c]: Yes, the caller can do that. But is that more friendly?

Comment (jforissier) [reply core/arch/arm/tee/entry_fast.c]: OK. Then I think I should also update "core: arm: GET_OS_REVISION: document a2 as a build identifier" and write that zero has a special meaning: "not specified", OK?

Comment (jenswi-linaro) [reply core/arch/arm/tee/entry_fast.c]: Sounds good

**Comment by jforissier** [comment]
Update

**Comment by jenswi-linaro** [comment]
`Reviewed-by: Jens Wiklander <jens.wiklander@linaro.org>`

**Pull request merged**